src/fastcs/datatypes.py

```diff
@@ -20,9 +21,40 @@ class DataType(Generic[T]):
     def dtype(self) -> type[T]:  # Using property due to lack of Generic ClassVars
         pass
 
+    @abstractmethod
+    def cast(self, value: T) -> Any:
```

Comment:
I don't think this should be called `cast` if `value` has to already be the correct type. In which case I am not sure the docstring is accurate.

I think all this is intended to do is check against its fields, e.g. `min`, `max`? Maybe `validate` would be more appropriate?

Reply:
For now I'll change it's name to `validate` (see below).